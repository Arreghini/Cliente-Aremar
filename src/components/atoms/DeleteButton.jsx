// src/components/atoms/DeleteButton.jsx
import React from 'react';
import { Trash2 } from 'lucide-react';
import PropTypes from 'prop-types';

const DeleteButton = ({ onDelete, reservationId }) => {
  return (
    <button
      onClick={() => onDelete(reservationId)}
      className="p-2 rounded-full hover:bg-red-100 text-red-600 transition-colors"
      title="Eliminar reserva"
    >
      <Trash2 className="w-5 h-5" />
    </button>
  );
};
DeleteButton.propTypes = {
  onDelete: PropTypes.func.isRequired,
  reservationId: PropTypes.string.isRequired,
};

export default DeleteButton;
