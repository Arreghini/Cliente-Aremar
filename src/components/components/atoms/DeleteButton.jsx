import React from 'react';
import reservationService from '../../services/ReservationService';
import { useAuth0 } from "@auth0/auth0-react";
import { FaTrash } from 'react-icons/fa';

const DeleteButton = ({ reservationId, onDelete }) => {
  const { getAccessTokenSilently } = useAuth0();

  const handleDelete = async () => {  
    if (window.confirm('¿Estás seguro de que deseas eliminar esta reserva?')) {
      try {
        const token = await getAccessTokenSilently();
        await reservationService.deleteReservation(token, reservationId);
        onDelete(reservationId);
      } catch (error) {
        console.error('Error al eliminar la reserva:', error);
      }
    }
  };
  return (
    <button onClick={() => handleDelete(reservationId)} className="text-red-500">
        <FaTrash />
    </button>
);
};

export default DeleteButton;
