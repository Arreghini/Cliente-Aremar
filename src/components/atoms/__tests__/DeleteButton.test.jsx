import { vi, describe, it, expect, beforeEach } from 'vitest'
import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import DeleteButton from '../DeleteButton'

describe('DeleteButton', () => {
  const mockOnDelete = vi.fn()
  const reservationId = '123ABC'

  beforeEach(() => {
    vi.clearAllMocks()
  })

  it('renderiza el botón con el ícono y el title correcto', () => {
    render(<DeleteButton onDelete={mockOnDelete} reservationId={reservationId} />)
    
    const button = screen.getByRole('button', { name: /Eliminar reserva/i })
    expect(button).toBeInTheDocument()
    
    // Verificar el title
    expect(button).toHaveAttribute('title', 'Eliminar reserva')
    
    // Verificar clases CSS
    expect(button).toHaveClass(
      'p-2 rounded-full hover:bg-red-100 text-red-600 transition-colors'
    )

    // Verificar que el ícono SVG está presente
    const icon = button.querySelector('svg')
    expect(icon).toBeInTheDocument()
  })

  it('llama a onDelete con reservationId al hacer click', () => {
    render(<DeleteButton onDelete={mockOnDelete} reservationId={reservationId} />)
    
    const button = screen.getByRole('button', { name: /Eliminar reserva/i })
    fireEvent.click(button)

    expect(mockOnDelete).toHaveBeenCalledTimes(1)
    expect(mockOnDelete).toHaveBeenCalledWith(reservationId)
  })

  it('llama a onDelete múltiples veces si se hace click varias veces', () => {
    render(<DeleteButton onDelete={mockOnDelete} reservationId={reservationId} />)
    
    const button = screen.getByRole('button', { name: /Eliminar reserva/i })
    fireEvent.click(button)
    fireEvent.click(button)
    fireEvent.click(button)

    expect(mockOnDelete).toHaveBeenCalledTimes(3)
    expect(mockOnDelete).toHaveBeenCalledWith(reservationId)
  })

  it('coincide con el snapshot', () => {
    const { asFragment } = render(
      <DeleteButton onDelete={mockOnDelete} reservationId={reservationId} />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
