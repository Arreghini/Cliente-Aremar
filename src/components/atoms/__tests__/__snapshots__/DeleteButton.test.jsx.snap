// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DeleteButton > coincide con el snapshot 1`] = `
<DocumentFragment>
  <button
    class="p-2 rounded-full hover:bg-red-100 text-red-600 transition-colors"
    title="Eliminar reserva"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-trash2 lucide-trash-2 w-5 h-5"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 6h18"
      />
      <path
        d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
      />
      <path
        d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
      />
      <line
        x1="10"
        x2="10"
        y1="11"
        y2="17"
      />
      <line
        x1="14"
        x2="14"
        y1="11"
        y2="17"
      />
    </svg>
  </button>
</DocumentFragment>
`;
